backends/ebpf/target.h

```diff
@@ -29,6 +29,17 @@ limitations under the License.
 
 namespace P4::EBPF {
 
+/*
+ * At various places, we'd like to test
+        dynamic_cast<const EBPF::P4TCTarget *>(builder->target)->isPrimitiveByteAligned(bits)
+ *  But, some of those places, we can't do that because we don't have a
+ *  builder yet, and isPrimitiveByteAligned() is a method on
```

Comment:
This sounds like a design flaw, what places are those? Maybe we have an easier solution here.

Reply:
I just checked the eBPF code and you have the builder available in all cases, so I am a little confused.

Reply:
Sorry, I'm probably missing something, but how would we access the `builder` in `ScanWidths::preorder` for the Assignment Statement [1]?

[1] https://github.com/vbnogueira/p4c/blob/128bit_ops_mouse/backends/tc/backend.cpp#L288

Reply:
In this case you know already that the target must be TC so you can just use the instance directly. The target just needs to be instantiated earlier and passed to the ScanWidths pass.